src/pages/images/actions/forms/UploadImageForm.tsx

```diff
@@ -0,0 +1,181 @@
+      const formData = new FormData();
+      const sortedFiles = Array.from(fileList).sort((a, b) => a.size - b.size);
+
+      formData.append("metadata", sortedFiles[0]);
+      formData.append("rootfs.img", sortedFiles[1]);
+
+      if (!formData.has("metadata") || !formData.has("rootfs.img")) {
+        throw new Error(
+          "Both metadata (tar.xz / gzip) and rootfs.img files are required",
+        );
+      }
+
+      return formData;
+    }
+    return fileList[0];
+  };
+
+  const formik = useFormik<{
+    alias?: string;
+    isPublic: boolean;
+    fileList: FileList | null;
+  }>({
+    initialValues: {
+      alias: "",
+      isPublic: false,
+      fileList: null,
+    },
+    validationSchema: Yup.object().shape({
+      alias: Yup.string(),
+    }),
+    onSubmit: (values) => {
+      if (values.fileList) {
+        createImageFromTarball(
+          getImageToTarballBody(values.fileList),
+          values.alias || "",
+          setUploadState,
+        )
+          .then((operation) => {
+            toastNotify.info(<>Creation of image from file started.</>);
+
+            eventQueue.set(
+              operation.metadata.id,
+              (event) => {
+                const fingerprint = event.metadata.metadata?.fingerprint ?? "";
+                if (values.alias) {
+                  void createImageAlias(fingerprint, values.alias);
+                }
+                void queryClient.invalidateQueries({
+                  queryKey: [queryKeys.images, project],
+                });
+                close();
+                notifySuccess();
+              },
+              (msg) => {
+                toastNotify.failure(`Image upload failed.`, new Error(msg));
+              },
+            );
+          })
+          .catch((e) => {
+            toastNotify.failure(`Image upload failed.`, e);
+          })
+          .finally(() => {
+            close();
+          });
+      }
+    },
+  });
+
+  return (
+    <Modal
+      close={close}
+      title="Import image from file"
+      className="create-image-from-tarball-modal"
+      buttonRow={
+        <>
+          {uploadState && (
+            <>
+              <ProgressBar percentage={Math.floor(uploadState.percentage)} />
+              <p>
+                {humanFileSize(uploadState.loaded)} loaded of{" "}
+                {humanFileSize(uploadState.total ?? 0)}
+              </p>
+              {uploadState.loaded === uploadState.total && (
+                <Loader text="Uploading file" />
+              )}
+            </>
+          )}
+          <Button
+            appearance="base"
+            className="u-no-margin--bottom"
+            type="button"
+            onClick={close}
+          >
+            Cancel
+          </Button>
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            loading={formik.isSubmitting}
+            disabled={!formik.isValid}
```

Comment:
I think we should disable submission if no files are selected
```suggestion
            disabled={!formik.isValid || !formik.values.fileList}
```